### PR TITLE
Fix: FSE and summary lines fixes

### LINKED ISSUE #849: LCFS - FSE input form refinements
**Description:**

Refinements for the Fuel Supplier Entry (FSE) system to improve user experience and data accuracy:

- Modify postal code validation to ignore spaces entered by the user. The error message should display as: “Please enter a valid postal code.”
- Update the duplicate function so that it does not duplicate the Serial number or Latitude/Longitude. Ensure validation checks that Serial numbers are unique.
- Adjust the default date of supply to avoid triggering an error, allowing users to proceed without selecting a date range. Ensure the default date range does not cause any errors.
- Ensure the Latitude/Longitude fields maintain consistent alignment. Keep the display aligned to the right even when editing, or left-align only if it’s the only way to maintain consistent alignment.
- Add predictive text functionality for the Manufacturer field using existing data.

**Purpose and Benefit to User:**

These changes will streamline data entry, reduce errors, and enhance the overall user experience in the FSE system. By improving validation, alignment, and adding predictive text, users will find it easier to enter accurate data quickly and efficiently.

**Acceptance Criteria:**

- [ ]  Given I am a user entering a postal code, when I type a space, then the system ignores the space and checks for a valid postal code, showing the error message “Please enter a valid postal code” only if it’s invalid.
- [ ]  Given I am a user using the duplicate function, when I duplicate an entry, then the Serial number and Lat/Long fields are not duplicated, and the system ensures that the Serial number remains unique.
- [ ]  Given I am a user working with the default date of supply, when I proceed without interacting with the fields, then the system does not trigger an error, and the default date range is saved as expected without issues.
- [ ]  Given I am a user editing the Latitude/Longitude fields, when I begin editing, then the alignment remains consistent (either right-aligned or left-aligned based on the chosen default).
- [ ]  Given I am a user entering the Manufacturer field, when I start typing, then predictive text suggestions appear based on existing data, helping me quickly select the correct entry.

**Development Checklist:**

 

- [x] Modify postal code validation to ignore spaces and update the error message.
- [x]  Update the duplicate function to exclude Serial number and Lat/Long from duplication and enforce unique Serial numbers.
- [ ]  Adjust the default date of supply to ensure no errors occur without selecting a date range, default full calendar year should be saved.
- [x]  Ensure consistent alignment for Latitude/Longitude fields during display and editing.
- [x]  Implement predictive text functionality for the Manufacturer field, pulling suggestions from existing FSE data.

### PR DESCRIPTION
**Pull request by prv-proton**
#849
fix for below
https://github.com/bcgov/lcfs/issues/849#issuecomment-2362347407
- Line 11 summary line fix
- line 17 fix https://github.com/bcgov/lcfs/issues/756#issuecomment-2349764800

### CODE REVIEW

File: backend/lcfs/web/api/transaction/repo.py [hunk 1]
```diff
@@ -264,7 +264,7 @@ async def calculate_available_balance_for_period(
             int: The available balance of compliance units for the specified organization and period. Returns 0 if no balance is calculated.
         """
         compliance_period_end = datetime.strptime(
-            f"{str(period + 1)}-12-31", "%Y-%m-%d"
+            f"{str(period + 1)}-03-31", "%Y-%m-%d"
```
Comment (dhaselhan) [inline backend/lcfs/web/api/transaction/repo.py:267]: This seems like a temporary fix, why do we need to manually set it to march?

**Comment by prv-proton** [comment]
> Cypress test report is available [here](https://github.com/bcgov/lcfs/actions/runs/10965481740). Download the 'cypress-results' artifact to view the report.

Thanks for the review @dhaselhan, that's the requirement set to calculate the transactions for the compliance period as per comment here
https://github.com/bcgov/lcfs/issues/756#issuecomment-2349764800.

please let me know if this needs any modifications or corrections.

File: backend/lcfs/web/api/transaction/repo.py (hunk 1 above)
Comment (prv-proton) [reply backend/lcfs/web/api/transaction/repo.py:267]: Thanks for the review @dhaselhan, that's the requirement set to calculate the transactions for the compliance period as per comment here
https://github.com/bcgov/lcfs/issues/756#issuecomment-2349764800.

please let me know if this needs any modifications or corrections.

Comment (dhaselhan) [reply backend/lcfs/web/api/transaction/repo.py:267]: ah okay, I see now. Its more that the year for the report isnt exactly a perfect year. Thanks for clarifying!

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Pull request merged**